---
title: "Stability of Vaginal microbiota during pregnancy and its importance for early infant microbiota using ASV"
output:
  html_document:
    toc: true
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = F, eval = F,warning = F,message = F,fig.width = 15, fig.height = 15, 
                      fig.align = 'center')
```

# Data

Data for this project is from the COPSAC2010 cohort of 711 children / mother pairs. 
In this project we describe the vaginal microbiome development from mid pregnancy (week 24) to late pregnancy (week 36), and the transfer to the airways and gut of the children in the first year of life. A special focus is on the differences between transfer to vaginal and sectio born children. 

```{r, eval=TRUE}
rm(list = ls())
library(tidyverse)
library(broom)
library(knitr)
# library("knitr", lib.loc="~/nas/R/x86_64-pc-linux-gnu-library/3.4")
library(ggrepel)
library(RColorBrewer)
library(cowplot)
# library(ggtree)
library(ggalluvial)
library(vegan)
library(phyloseq)
# load('COPSACbirthmicrobiome_v2.RData')
load('COPSACbirthmicrobiome_ASV.RData')
```

```{r, eval=FALSE}
### setting up the ASV data to match the formating of the initial ASV based analysis
load('/nas/data/share/projects/DADA2golden/all.2k.dada2.golden.Rdata')
load('../AnonymizationKey.RData')
delivery <- rio::import('../ABC0016_Delivery.xlsx',sheet = 'Data')
x <-subset_samples(all.2k.dada2.golden, Time %in% c('1m','1w','1y','24','36','3m'))
x = filter_taxa(x, function(x) sum(x) > 0, TRUE)

# LinkingABC2Dyad <- data.frame(abc = 2000:2900, dyadnb = as.character(paste('dyad',str_pad(sample(1:901,replace = F),3,pad = '0'), sep = '')))

# set sample data
SD <- sample_data(x) %>% 
  select(Abcno, Time, Type) %>% 
  data.frame() %>%
  mutate(abc = Abcno %>% as.character %>% as.numeric ) %>%
  left_join(delivery, by = c('abc' = 'Abcno')) %>% 
  rename('DELIVERY' = 'Delivery') %>% 
  left_join(LinkingABC2Dyad, by = c('abc' = 'abc')) %>%
  mutate(samplename = paste(dyadnb,Type,Time,sep = '_')) %>% 
  select(dyadnb, Time, Type, DELIVERY, samplename)
row.names(SD) <- SD$samplename

OTUmat <- otu_table(x)
dimnames(OTUmat)[[2]] <- SD$samplename

SD <- SD %>% select(-samplename)

phyX <- phyloseq(OTUmat,sample_data(SD),tax_table(x),phy_tree(x))

save(phyX,file ='COPSACbirthmicrobiome_ASV.RData')
```


# Descriptives

The microbiome make up is described here overall. 

## Size, number of reads and sparsity

```{r,eval=TRUE}
SD <- sample_data(phyX)
tb0 <- table(SD$Type,SD$Time)
rowSums(tb0)

tb <- table(SD$dyadnb,SD$Type) %>% as.data.frame.matrix()
dim(tb)

sum(tb$V>0)

tbpair <- tb[tb$V>0 & (tb$F> 0 | tb$T>0),]
dim(tbpair)


print(phyX)

fecX <- subset_samples(phyX,Type=='F')
airX <- subset_samples(phyX,Type=='T')
vagX <- subset_samples(phyX,Type=='V')

df_type_stat <- data.frame(
  nfec =  apply(otu_table(fecX)>0,1,sum),
  nair =  apply(otu_table(airX)>0,1,sum),
  nvag =  apply(otu_table(vagX)>0,1,sum))
apply(df_type_stat>0,2,sum)
# get number of identified ASV's in each compartment


df <- data.frame(depth = sample_sums(phyX),
                 nobserved = apply(otu_table(phyX)>0,2,sum),
                 sample_data(phyX))
ggplot(data = df, aes(x = depth, fill = Type:Time)) + 
  geom_histogram()+ 
  scale_x_log10() + 
  facet_wrap(~Type:Time)

tb <- df %>% group_by(Type,Time) %>% 
  summarise(median_count = median(depth), 
            mean_count = mean(depth), 
            sd_count = sd(depth), 
            q25_count= quantile(depth)[2],
            q75_count= quantile(depth)[4], 
            median_observed = median(nobserved),
            mean_observed = mean(nobserved), 
            sd_observed = sd(nobserved), 
            min_observed = min(nobserved), 
            max_observed = max(nobserved)
  )

kable(tb,caption = 'Summary stats for compartment/timepoint', digits = 0)
```

# Vaginal descriptives

The distribution of the vaginal reads are here summarized on phylum, family and individual ASV level. 

```{r}
vagX <- subset_samples(phyX,Type=='V')
vagXphy <- tax_glom(vagX,'Phylum')
vagXgenus <- tax_glom(vagX,'Genus_simple')
infX <- subset_samples(phyX,Type!='V')
save(file = './vag_taxglm.RData',list = c('vagX','vagXphy','vagXgenus','infX'))
```

Observed vaginal ASV's: 

```{r,eval=TRUE}
load('./vag_taxglm.RData')
# print(vagXphy)
# print(vagXgenus)

print(sum(apply(otu_table(vagX)>0,1,sum)>0))

df2 <- data.frame(tax_table(vagXphy),taxSum = taxa_sums(vagXphy))
df2$taxprc <- 100*df2$taxSum/sum(df2$taxSum)
df3 <- data.frame(tax_table(vagXgenus),taxSum = taxa_sums(vagXgenus))
df3$taxprc <- 100*df3$taxSum/sum(df3$taxSum)
df4 <- data.frame(tax_table(vagX),taxSum = taxa_sums(vagX))
df4$taxprc <- 100*df4$taxSum/sum(df4$taxSum)

# Top 3 dominating phyla prc reads "Firmicutes (85.0%), Acinobacteria (11.8%) and Proteobacteria (2.0%)"
kable(head(df2[order(df2$taxSum,decreasing = T),]), row.names = F,digits = 1, caption = 'Distribution of reads according to phylym')
# Top 2 dominating genus prc reads "Lactobacillus (78.5%) and Gardnerella (8.7%)"
kable(head(df3[order(df3$taxSum,decreasing = T),]), row.names = F,digits = 1, caption = 'Distribution of reads according to genus')
# Top 4 Lactobacilli  "The most abundant lactobacilli were L. crispatus (33.3%), L. iners (28.6%), L. gasseri (10.7%), and L. jensenii (4.9%). ""
kable(head(df4[order(df4$taxSum,decreasing = T),]), row.names = F,digits = 1, caption = 'Distribution of reads according to ASV')
```


## Community State Types

The vaginal microbiome is _not_ a smooth continoum, but merely a partition between several microbiome-phenotypes, here refered to as Community State Types (CST). These are identified by clustering of _all_ the samples based on Jensen Shannon Divergence as beta diversity measure. The dendrogram is cutted to reveal _6_ clusters. These are refered to as community state types I to VI.   

```{r}
load('./vag_taxglm.RData')
vag.active <- prune_taxa(taxa_sums(vagX) > 0, vagX)
vag.active.r <- rarefy_even_depth(vag.active, 2000, rngseed = 2)
vag.transformed <- transform_sample_counts(vag.active.r, function(x) x/sum(x))
vag.jsd <- distance(vag.transformed, method="jsd")

clust <- hclust(vag.jsd, method = "ward.D2")
sample_data(vagX)$CST <- cutree(clust, k=6)

sample_data(vagX)$CST <- as.factor(sample_data(vagX)$CST)
levels(sample_data(vagX)$CST) <- list(CST_I = 3, CST_II = 1, CST_III = 2, CST_IV_b = 5, CST_IV_c = 4, CST_V = 6)

# table with top dominating taxa / ASVs in each CST

# Phylum
df2 <- data.frame(otu_table(vagXphy) %>% t(), sample_data(vagX)) 
txtb <- tax_table(vagXphy) %>% as.data.frame()
txtb$ASV <- rownames(txtb)
tb0 <- df2 %>%
  group_by(dyadnb) %>%
  mutate(rep = n()) %>%
  ungroup() %>%
  group_by(CST) %>%
  summarise(n = n(), n_w_rep = sum(rep==2),
            n24 = sum(Time==24),n36 = sum(Time==36))

tb1 <- df2 %>%
  gather(ASV,cnt, X4c663e2f2c9a3f1d1cb52bd57cf0382d:X6fc334e897e5ada6a360dd2bef0c3eab) %>%
  left_join(txtb, by = 'ASV') %>%
  group_by(dyadnb,Time) %>%
  mutate(libsize = sum(cnt), 
         cnt = cnt / libsize) %>%
  ungroup() %>%
  group_by(CST,Phylum) %>%
  summarise(totcnt = mean(cnt)*100) %>%
  ungroup() %>%
  group_by(CST) %>%
  mutate(rnk = rank(totcnt), 
         rnk = max(rnk)-rnk+1) %>%
  arrange(CST,desc(totcnt)) %>%
  filter(rnk<6)

# Genus
df2 <- data.frame(otu_table(vagXgenus) %>% t(), sample_data(vagX)) 
txtb <- tax_table(vagXgenus) %>% as.data.frame()
txtb$ASV <- rownames(txtb)
tb1g <- df2 %>%
  gather(ASV,cnt,X8a5de2b351d0e097a9174803a5cdaf75:fccf9b38f0596644eab0612a5cd57597) %>%
  left_join(txtb, by = 'ASV') %>%
  group_by(dyadnb,Time) %>%
  mutate(libsize = sum(cnt), 
         cnt = cnt / libsize) %>%
  ungroup() %>%
  group_by(CST,Genus) %>%
  summarise(totcnt = mean(cnt)*100) %>%
  ungroup() %>%
  group_by(CST) %>%
  mutate(rnk = rank(totcnt), 
         rnk = max(rnk)-rnk+1) %>%
  arrange(CST,desc(totcnt)) %>%
  filter(rnk<6)

# ASV

ASVtab <- otu_table(vagX) %>% t()
# top 100 overall
ss <- rank(apply(ASVtab,2,sum))
ss <- max(ss) - ss + 1
df2 <- data.frame(ASVtab[,ss<100], sample_data(vagX)) 
txtb <- tax_table(vagX) %>% as.data.frame()
txtb$ASV <- rownames(txtb)
tb1ASV <- df2 %>%
  gather(ASV,cnt,f15b68147a589c20192c2cc8cc518f0d:b4f36013dae7d7ba0ceeaafb5842c6d9) %>%
  left_join(txtb, by = 'ASV') %>%
  group_by(dyadnb,Time) %>%
  mutate(libsize = sum(cnt), 
         cnt = cnt / libsize) %>%
  ungroup() %>%
  group_by(CST,ASV) %>%
  summarise(totcnt = mean(cnt)*100) %>%
  ungroup() %>%
  group_by(CST) %>%
  mutate(rnk = rank(totcnt), 
         rnk = max(rnk)-rnk+1) %>%
  arrange(CST,desc(totcnt)) %>%
  filter(rnk<6)


colnames(tb1)[3] <- 'Phylum_prc'
colnames(tb1g)[3] <- 'Genus_prc'
colnames(tb1ASV)[3] <- 'ASV_prc'
tb1m <- merge(merge(tb1,tb1g, by = c('CST','rnk')), tb1ASV,by = c('CST','rnk'))

save(file = './CommunityStateTypes.RData',list = c('vagX','clust','vag.jsd','tb0','tb1m'))
```

### Figure S1 - Alluvial plot with CST

```{r,eval=TRUE}
load('./CommunityStateTypes.RData')

kable(tb0, caption = 'Sample Distribution, n: number of samples in CST, n_w_rep: number of samples in CST from women with both timepoints represented, n24 / n36: number of samples from week 24 and 36 respectively')

kable(tb1m, caption = 'Top five Phylums / Genus / ASVs for each CST', digits = 2)

gAL <- sample_data(vagX) %>%
  ggplot(data = .,
         aes(x = Time, stratum = CST, alluvium = dyadnb,
             # weight = 1,
             fill = CST, label = CST)) +
  geom_flow() +
  scale_fill_brewer(palette = 'Set1') + 
  scale_x_discrete(expand = c(0,0)) + 
  geom_stratum(alpha = .5) +
  # facet_grid(cleandef~trt) + 
  geom_text(stat = "stratum", size = 3) +
  theme_bw() + 
  xlab('Pregnancy week') + 
  theme(legend.position = 'none',panel.grid = element_blank(), panel.border = element_blank())

print(gAL)

pdf('FigureS1.pdf',height = 8, width = 4)
gAL
dev.off()
```

```{r}
load('./vag_taxglm.RData')
load('./CommunityStateTypes.RData')

vag.active <- prune_taxa(taxa_sums(vagX) > 0, vagX)

cl <- parallel::makeCluster(20)
doParallel::registerDoParallel(cl)
vag.WUnifrac <- UniFrac(vag.active, weighted=TRUE, parallel = TRUE)
vag.all.nmds <- metaMDS(vag.WUnifrac, k = 4, trymax = 100)
vag.all.jsd.nmds <- metaMDS(vag.jsd, k = 5, trymax = 100)

save(file = './OrdinationRes.RData',list = c('vag.WUnifrac','vag.all.nmds','vag.all.jsd.nmds'))
```

## Ordination of of vag samples with CTS and timepoints

### Figure S2

Here PCoA plots show the distirbution of the samples based on CST and beta diversity metric. 
Clearly, some of the CST are more well defined than others. E.g. CST_IV_b and CST_IV_c are all over the place. 

```{r,eval=T}

g_legend<-function(a.gplot){ 
  tmp <- ggplot_gtable(ggplot_build(a.gplot)) 
  leg <- which(sapply(tmp$grobs, function(x) x$name) == "guide-box") 
  legend <- tmp$grobs[[leg]] 
  return(legend)} 
mytheme <- theme(axis.title = element_text(size = 15), axis.text = element_text(size = 10)) + theme_bw()

load('./vag_taxglm.RData')
load('./CommunityStateTypes.RData')
load('./OrdinationRes.RData')

# df <- rbind(data.frame(method = 'wunifrac', sample_data(vagX),vag.all.nmds$points, MDS5 = 0),
# data.frame(method = 'jsd', sample_data(vagX),vag.all.jsd.nmds$points))

df <- data.frame(method = 'jsd', sample_data(vagX),vag.all.jsd.nmds$points)


# PCoA with CST 1 vs 2
g1 <- ggplot(data = df, aes(MDS1,MDS2, color = CST,group= dyadnb)) + 
  stat_ellipse(aes(group = CST)) + 
  geom_line(color = 'grey90') + 
  geom_point() + 
  scale_color_brewer(palette = 'Set1') + 
  # facet_wrap(~method, scales = 'free') + 
  theme(legend.position = 'none')

# PCoA with CST 3 vs 4
g2 <- ggplot(data = df, aes(MDS3,MDS4, color = CST,group= dyadnb)) + 
  stat_ellipse(aes(group = CST)) + 
  geom_line(color = 'grey90') + 
  geom_point() + 
  scale_color_brewer(palette = 'Set1') + 
  # facet_wrap(~method, scales = 'free') + 
  theme(legend.position = 'bottom')


legend <- g_legend(g2)
# G1 <- ggdraw() + draw_plot(g1 +  mytheme + theme(legend.position = "hidden"), 0, .55, 1, .45) + 
# draw_plot(g2 + mytheme+ theme(legend.position = "hidden"), 0, 0.1, 1, .45)+
# draw_grob(legend, 0, 0, 1, .1) 
G1 <- ggdraw() + draw_plot(g1 +  mytheme + theme(legend.position = "hidden"), 0, 0.1, 0.5, 0.9) + 
  draw_plot(g2 + mytheme+ theme(legend.position = "hidden"), 0.5, 0.1, 0.5, 0.9)+
  draw_grob(legend, 0, 0, 1, .1) 

print(G1)

pdf('FigureS2.pdf', width = 10, height = 5)
print(G1)
dev.off()


```

## Stability between w24 and w36

In order to test the stability between time points, a permutation procedure is used. Here, the distance based on individual women from week 24 to week 36 are compared with random assignments of pairs.  

```{r}
nperm <- 2500
# @MARTIN can you plug in the code for the two tests. 
# Likelihood of changing between clusters (#5) 
# Ratio of Jensen-Shannon Divergence (only on women in pairs) - median ratio = 0.159 

# get dist for all pairs
getDist <- function(x,D){
  # D <- as.matrix(vag.WUnifrac)
  #_# x <- sd_vagX %>% filter(dyadnb=='dyad219')
  id <- rownames(D) %in% x$smpname
  if (sum(id)==2){
    dd <- D[id,id]
    dst <- dd[1,2]
  }
  else {dst <- NA}
  return(data.frame(distw24_w36 = dst))
}

sd_vagX <- sample_data(vagX) 
sd_vagX$smpname = rownames(sd_vagX)

# Wunifrac
Dwuf <- as.matrix(vag.WUnifrac)

ddfmodel <- sd_vagX %>%
  group_by(dyadnb) %>%
  do(getDist(x = ., Dwuf)) %>%
  left_join(sd_vagX[sd_vagX$Time==24,], by = 'dyadnb')

ddfperm  <- c()
for (i in 1:nperm){
  print(i)
  sd_vagX$dyadnb2 <- sd_vagX$dyadnb
  sd_vagX$dyadnb2[sd_vagX$Time==36] <- sample(sd_vagX$dyadnb2[sd_vagX$Time==36])
  ddf2 <- sd_vagX %>%
    group_by(dyadnb2) %>%
    do(getDist(x = ., Dwuf)) %>%
    mutate(permutation = i)
  ddfperm <- rbind(ddfperm,ddf2)
}

# calculate the statistics
pv <- ddfperm %>%
  left_join(ddfmodel, by = c('dyadnb2' = 'dyadnb')) %>%
  filter(!is.na(distw24_w36.x) & !is.na(distw24_w36.y)) %>%
  group_by(dyadnb2) %>%
  summarise(n = n(), 
            npos = sum(distw24_w36.x < distw24_w36.x), 
            freq = npos / n) %>%
  ungroup %>%
  summarise(pv = mean(freq))

dfmodelwuf <- ddfmodel
dfpermwuf <- ddfperm
pv_wuf <- pv

# JSD model
Djsd <- as.matrix(vag.jsd)

ddfmodel <- sd_vagX %>%
  group_by(dyadnb) %>%
  do(getDist(x = ., Djsd)) %>%
  left_join(sd_vagX[sd_vagX$Time==24,], by = 'dyadnb')

ddfperm  <- c()
for (i in 1:nperm){
  print(i)
  sd_vagX$dyadnb2 <- sd_vagX$dyadnb
  sd_vagX$dyadnb2[sd_vagX$Time==36] <- sample(sd_vagX$dyadnb2[sd_vagX$Time==36])
  ddf2 <- sd_vagX %>%
    group_by(dyadnb2) %>%
    do(getDist(x = ., Djsd)) %>%
    mutate(permutation = i)
  ddfperm <- rbind(ddfperm,ddf2)
}

# calculate the statistics
pv <- ddfperm %>%
  left_join(ddfmodel, by = c('dyadnb2' = 'dyadnb')) %>%
  filter(!is.na(distw24_w36.x) & !is.na(distw24_w36.y)) %>%
  group_by(dyadnb2) %>%
  summarise(n = n(), 
            npos = sum(distw24_w36.x < distw24_w36.x), 
            freq = npos / n) %>%
  ungroup %>%
  summarise(pv = mean(freq))

dfmodeljsd <- ddfmodel
dfpermjsd <- ddfperm
pv_jsd <- pv


save(file = './Stability_w24_to_2w6_permresults.RData',
     list = c('dfmodelwuf','dfpermwuf','pv_wuf','dfmodeljsd','dfpermjsd','pv_jsd'))
```

### Tables with stability results

```{r, eval = TRUE}
load('./Stability_w24_to_2w6_permresults.RData')

sd_vagX <- sample_data(vagX) 
# remove single tons
ttb <- sd_vagX %>%
  group_by(dyadnb) %>%
  mutate(n = n()) %>%
  filter(n==2) %>%
  group_by(dyadnb) %>%
  mutate(insame = ifelse(length(unique(CST))==1,1,0)) %>%
  group_by(CST,Time) %>%
  summarise(n = n(), 
            n_insame = sum(insame), 
            prc_insame = 100*n_insame/n)

ntot <- sum(ttb$n)/2

ttb2 <- ttb %>% 
  filter(Time==24) %>%
  ungroup %>%
  summarise(n = sum(n_insame), 
            prc = 100*n / ntot)

kable(ttb2,caption = 'overall stability descriptives from week 24 and w 36 dependent', digits = 1)

kable(ttb,caption = 'CST descriptives from week 24 and w 36 dependent', digits = 1)

dfwuf <- dfpermwuf %>%
  left_join(dfmodelwuf, by = c('dyadnb2' = 'dyadnb')) 
dfjsd <- dfpermjsd %>%
  left_join(dfmodeljsd, by = c('dyadnb2' = 'dyadnb')) 


DF <- bind_rows(data.frame(dfwuf,method = 'wuf'),data.frame(dfjsd,method = 'jsd')) #%>%

tb <- DF %>%
  filter(!is.na(distw24_w36.y) & permutation==1) %>%
  group_by(method) %>%
  summarise(n= n(), median_dist = median(distw24_w36.y, na.rm = T))

tb_CST <- DF %>%
  filter(!is.na(distw24_w36.y) & permutation==1) %>%
  group_by(method,CST) %>%
  summarise(n= n(), median_dist = median(distw24_w36.y, na.rm = T))

tb_CSTpv <- DF %>%
  filter(!is.na(distw24_w36.y) & permutation==1) %>%
  group_by(method) %>%
  do(kruskal.test(data = .,distw24_w36.y~CST) %>% tidy %>% select(-method))


tb2 <- DF %>%
  filter(!is.na(distw24_w36.x) & !is.na(distw24_w36.y)) %>%
  group_by(method, permutation) %>%
  summarise(median_permdist = median(distw24_w36.x)) %>%
  ungroup %>%
  left_join(tb) %>%
  group_by(method) %>%
  summarise(
    n = n[1],
    median_dist = median_dist[1], 
    median_permdist = mean(median_permdist),
    R = median_permdist / median_dist,
    pv = sum(1+(median_permdist<median_dist)) / (1+n()))

kable(tb2, caption = 'Stability between week 24 and w 36 assigned as median distance between pairs as compared with mismatched pairs (# permutations = 2500)', digits = 5)

kable(tb_CST,caption = 'Stability between week 24 and w 36 dependent on week 24 CST', digits = 4)

ggplot(data = tb_CST, aes(CST, median_dist, fill = CST)) + 
  geom_bar(stat = 'identity') +
  scale_fill_brewer(palette = 'Set1') + 
  facet_wrap(~method, scales = 'free') + 
  theme_bw() + 
  theme(legend.position = 'none')

kable(tb_CSTpv,caption = 'Inference (kruskal walis) for differences in stability between week 24 and w 36 dependent on week 24 CST', digits = 100)
```

# Infant descriptives

```{r,eval=T}

infant <- sample_data(phyX) %>%
  data.frame() %>%
  filter(!is.na(DELIVERY)) %>%
  filter(!duplicated(dyadnb))

rbind(table(infant$DELIVERY) ,
      100*table(infant$DELIVERY) / dim(infant)[1])

vagdyad <- sample_data(vagX)$dyadnb[sample_data(vagX)$Time == '36']

tb2 <- infX %>% 
  sample_data %>% 
  group_by(Type, Time,DELIVERY) %>% 
  summarise(n = n(), 
            n_with_vagw36 = sum(dyadnb %in% vagdyad  ), 
            prc = 100*n_with_vagw36 / n) %>%
  data.frame()
kable(tb2, digits = 1, caption = 'Number of infant samples (n), with a corresponding maternal vaginal w36 sample (n_with_vagw36)')
```

The statistics indicate that stability depends on CST. 

# Transfer

## Functions for transfer models

Here, we have a bunch of internal function with specific algorithms. 

```{r,eval=TRUE}
gt <- function(x){
  tb <- table(x$countC>0,x$count01) 
  dtb <- tb %>% dim()
  tb <- data.frame(n1 = dtb[1],n2 = dtb[2],nZero = sum(tb==0))
  return(tb)
}

getFisher <- function(x,doglm = TRUE){
  # x <- STS[1,1:5]
  # doglm = TRUE
  x$or <- (x$n00*x$n11) / (x$n10*x$n01)
  side <- ifelse(x$or<1,'less','greater')
  nn <- x$n00 + x$n10 + x$n01 + x$n11
  tb <- x[,c('n00','n01','n10','n11')] %>% 
    unlist() %>%
    matrix(2) %>%
    fisher.test(alternative = side) %>%
    tidy
  colnames(tb) <- paste('Fisher',colnames(tb),sep = '_')
  
  # bias corrected
  or_biascorr <- (x$n00 + 0.5)*(x$n11+0.5) / (x$n01 + 0.5)*(x$n10+0.5)
  t_biascorr <- (nn*(abs(x$n00*x$n11 - x$n10*x$n01) - 0.5*nn)^2) / ((x$n00 + x$n01)*(x$n00 + x$n10)*(x$n10 + x$n11)*(x$n01 + x$n11))
  pv_biascorr <- 2*(1 - pnorm(t_biascorr))
  biascorr <- data.frame(or_biascorr,t_biascorr,pv_biascorr)
  
  # Gtest accoring to 
  # gtest <- x[,c('n00','n01','n10','n11')] %>% 
    # unlist() %>%
    # matrix(2) %>%
    # DescTools::GTest(correct = 'yates') %>%
    # DescTools::GTest() %>% 
    # tidy
  # colnames(gtest) <- paste('Gtest',colnames(gtest),sep = '_')
  # tb <- cbind(tb,biascorr,gtest)
  
  tb <- cbind(tb,biascorr)
  
  # glm
  if (doglm){
    mom <- c(rep(0,x$n00), rep(0,x$n01), rep(1,x$n10),rep(1,x$n11))
    child <- c(rep(0,x$n00), rep(1,x$n01), rep(0,x$n10),rep(1,x$n11))
    mdl <- glm(child~mom,family = binomial('logit')) %>% 
      tidy %>%
      filter(term=='mom') %>%
      mutate(or = exp(estimate)) %>%
      select(-term)
    colnames(mdl) <- paste('Glm',colnames(mdl),sep = '_')
    tb <- cbind(tb,mdl)
  } 
  return(tb)
}

truncateZerosInf <- function(or,trc = 100){
  # trc = 100
  # or <- c(Inf,0.01,1,10,200,-Inf)
  trcm <- 10^-log10(trc)
  ornew <- or
  ornew[(is.infinite(or) & or>1) | or>trc ] <- trc
  ornew[(or==0 & or<1) | or<trcm] <- trcm
  return(ornew)
}

getWeigtedRatio <- function(x,e = 0.001){
  or <- x$Fisher_estimatetr
  # pv <- 1-x$Fisher_p.value
  pv <- -log10(x$Fisher_p.value)
  # pv <- -log10(x$Gtest_p.value)
  np <- sum(or>=1)
  nn <- sum(or<1)
  # mass
  area <- log(or)*pv
  ratio <- (sum(area[area>=0]) + e)/(-sum(area[area<0]) + e)
  R <- data.frame(np,nn , ratio)
  return(R)
}

shuffle <- function(nestedfactor){
  # nestedfactor <- sd1$delivery
  id <- 1:length(nestedfactor)
  df <- data.frame(nestedfactor,id,idnew=id)
  unf <- unique(df$nestedfactor)
  unf <- unf[!is.na(unf)]
  
  for (i in unf){
    ic <- df$nestedfactor==i & !is.na(df$nestedfactor)
    ids <- df$id[ic]
    df$idnew[ic] <- sample(ids)
  }
  ic <- is.na(df$nestedfactor)
  ids <- df$id[ic]
  df$idnew[ic] <- sample(ids)
  idnew <- df$idnew
  return(idnew)
}

getWeigtedRatio2 <- function(x,e = 0.001){
  or <- x$or_biascorr
  # pv <- 1-x$Fisher_p.value
  pv <- -log10(x$pv_biascorr)
  np <- sum(or>=1)
  nn <- sum(or<1)
  # mass
  area <- log(or)*pv
  ratio <- (sum(area[area>0]) + e)/(-sum(area[area<=0]) + e)
  R <- data.frame(np,nn , ratio)
  return(R)
}


extractPV <- function(permSTAT,modelratio,trm=100){
  # modelratio <- tb$ratio[5]
  # permSTAT <- permSTAT_w36_F1w_norm  
  # modelratio <- 2.5
  # trm <- 100
  
  
  niter <- dim(permSTAT)[3]
  tb <- c()
  for (i in 1:niter){
    xx <- permSTAT[,,i] %>%
      data.frame() %>%
      mutate(Fisher_estimatetr = truncateZerosInf(Fisher_estimate,trm)) %>%
      do(getWeigtedRatio(x = .))
    tb <- rbind(tb,xx)
  }
  pv <- sum(tb$ratio>modelratio) / niter
  
  # estimate null distribution for ratio
  lgratio <- log(tb$ratio)
  SElgratio <- sqrt(sum(lgratio^2)/length(lgratio))
  # 1 - pnorm(modelratio / SElgratio)
  
  # calculate SE for model ratio
  # qnorm(pv)
  
  print(c(i,median(tb$ratio),modelratio))
  df <- data.frame(pv, SElgratio, permmedian = median(tb$ratio), modelratio)
  
  return(df)
}


# "Transfer_Uni_dico_w36.pdf"
# Number of testable ASVs: "420 and 527 ASVs obtained a dynamic range in both mother and child for transfer to hypopharyngeal and fecal samples"
# Vulcano no dico-odds for tranfer
# Statistics on proportion of positive odds:  "factor of 1.8 (1.3-2.6) and 3.9 (2.8-5.4) (p < 0.001 for both) for hypopharyngeal and fecal samples"
```

## ALL individual ASV models

Here all the combinations are calculated. 

```{r}
nperm <-1000
######### VAGINAL BORN
# w36 vs F week 1
phy1 <- phyX %>% subset_samples(Type=='V' & Time == '36' & DELIVERY=='Normal')
phy2 <- phyX %>% subset_samples(Time == '1w' & Type == 'F' & DELIVERY=='Normal') 



source('getTransferStats.R')
STAT_w36_F1w_norm <- STAT
permSTAT_w36_F1w_norm <- permSTATfisher

# w36 vs F month 1
phy2 <- phyX %>% subset_samples(Time == '1m' & Type == 'F' & DELIVERY=='Normal') 
source('getTransferStats.R')
STAT_w36_F1m_norm <- STAT
permSTAT_w36_F1m_norm <- permSTATfisher

# w36 vs F year 1
phy2 <- phyX %>% subset_samples(Time == '1y' & Type == 'F' & DELIVERY=='Normal') 
source('getTransferStats.R')
STAT_w36_F1y_norm <- STAT
permSTAT_w36_F1y_norm <- permSTATfisher

# w36 vs T week 1
phy2 <- phyX %>% subset_samples(Time == '1w' & Type == 'T' & DELIVERY=='Normal') 
source('getTransferStats.R')
STAT_w36_T1w_norm <- STAT
permSTAT_w36_T1w_norm <- permSTATfisher

# w36 vs T month 1
phy2 <- phyX %>% subset_samples(Time == '1m' & Type == 'T' & DELIVERY=='Normal') 
source('getTransferStats.R')
STAT_w36_T1m_norm <- STAT
permSTAT_w36_T1m_norm <- permSTATfisher

# w36 vs T month 3
phy2 <- phyX %>% subset_samples(Time == '3m' & Type == 'T' & DELIVERY=='Normal') 
source('getTransferStats.R')
STAT_w36_T3m_norm <- STAT
permSTAT_w36_T3m_norm <- permSTATfisher

######### C-sectio BORN
# w36 vs F week 1
phy1 <- phyX %>% subset_samples(Type=='V' & Time == '36' & DELIVERY!='Normal')
phy2 <- phyX %>% subset_samples(Time == '1w' & Type == 'F' & DELIVERY!='Normal') 
source('getTransferStats.R')
STAT_w36_F1w_csec <- STAT
permSTAT_w36_F1w_csec <- permSTATfisher

# w36 vs F month 1
phy2 <- phyX %>% subset_samples(Time == '1m' & Type == 'F' & DELIVERY!='Normal') 
source('getTransferStats.R')
STAT_w36_F1m_csec <- STAT
permSTAT_w36_F1m_csec <- permSTATfisher

# w36 vs F year 1
phy2 <- phyX %>% subset_samples(Time == '1y' & Type == 'F' & DELIVERY!='Normal') 
source('getTransferStats.R')
STAT_w36_F1y_csec <- STAT
permSTAT_w36_F1y_csec <- permSTATfisher

# w36 vs T week 1
phy2 <- phyX %>% subset_samples(Time == '1w' & Type == 'T' & DELIVERY!='Normal') 
source('getTransferStats.R')
STAT_w36_T1w_csec <- STAT
permSTAT_w36_T1w_csec <- permSTATfisher

# w36 vs T month 1
phy2 <- phyX %>% subset_samples(Time == '1m' & Type == 'T' & DELIVERY!='Normal') 
source('getTransferStats.R')
STAT_w36_T1m_csec <- STAT
permSTAT_w36_T1m_csec <- permSTATfisher

# w36 vs T month 3
phy2 <- phyX %>% subset_samples(Time == '3m' & Type == 'T' & DELIVERY!='Normal') 
source('getTransferStats.R')
STAT_w36_T3m_csec <- STAT
permSTAT_w36_T3m_csec <- permSTATfisher

######### C-sectio - planned 
# w36 vs F week 1
phy1 <- phyX %>% subset_samples(Type=='V' & Time == '36' & DELIVERY=='Planned sectio')
phy2 <- phyX %>% subset_samples(Time == '1w' & Type == 'F' & DELIVERY=='Planned sectio') 
source('getTransferStats.R')
STAT_w36_F1w_csec_planned <- STAT
permSTAT_w36_F1w_csec_planned <- permSTATfisher

# w36 vs F month 1
phy2 <- phyX %>% subset_samples(Time == '1m' & Type == 'F' & DELIVERY=='Planned sectio') 
source('getTransferStats.R')
STAT_w36_F1m_csec_planned <- STAT
permSTAT_w36_F1m_csec_planned <- permSTATfisher

# w36 vs F year 1
phy2 <- phyX %>% subset_samples(Time == '1y' & Type == 'F' & DELIVERY=='Planned sectio') 
source('getTransferStats.R')
STAT_w36_F1y_csec_planned <- STAT
permSTAT_w36_F1y_csec_planned <- permSTATfisher

# w36 vs T week 1
phy2 <- phyX %>% subset_samples(Time == '1w' & Type == 'T' & DELIVERY=='Planned sectio') 
source('getTransferStats.R')
STAT_w36_T1w_csec_planned <- STAT
permSTAT_w36_T1w_csec_planned <- permSTATfisher

# w36 vs T month 1
phy2 <- phyX %>% subset_samples(Time == '1m' & Type == 'T' & DELIVERY=='Planned sectio') 
source('getTransferStats.R')
STAT_w36_T1m_csec_planned <- STAT
permSTAT_w36_T1m_csec_planned <- permSTATfisher

# w36 vs T month 3
phy2 <- phyX %>% subset_samples(Time == '3m' & Type == 'T' & DELIVERY=='Planned sectio') 
source('getTransferStats.R')
STAT_w36_T3m_csec_planned <- STAT
permSTAT_w36_T3m_csec_planned <- permSTATfisher

######### C-sectio - Acute
# w36 vs F week 1
phy1 <- phyX %>% subset_samples(Type=='V' & Time == '36' & DELIVERY=='Acute sectio')
phy2 <- phyX %>% subset_samples(Time == '1w' & Type == 'F' & DELIVERY=='Acute sectio') 
source('getTransferStats.R')
STAT_w36_F1w_csec_acute <- STAT
permSTAT_w36_F1w_csec_acute <- permSTATfisher

# w36 vs F month 1
phy2 <- phyX %>% subset_samples(Time == '1m' & Type == 'F' & DELIVERY=='Acute sectio') 
source('getTransferStats.R')
STAT_w36_F1m_csec_acute <- STAT
permSTAT_w36_F1m_csec_acute <- permSTATfisher

# w36 vs F year 1
phy2 <- phyX %>% subset_samples(Time == '1y' & Type == 'F' & DELIVERY=='Acute sectio') 
source('getTransferStats.R')
STAT_w36_F1y_csec_acute <- STAT
permSTAT_w36_F1y_csec_acute <- permSTATfisher

# w36 vs T week 1
phy2 <- phyX %>% subset_samples(Time == '1w' & Type == 'T' & DELIVERY=='Acute sectio') 
source('getTransferStats.R')
STAT_w36_T1w_csec_acute <- STAT
permSTAT_w36_T1w_csec_acute <- permSTATfisher

# w36 vs T month 1
phy2 <- phyX %>% subset_samples(Time == '1m' & Type == 'T' & DELIVERY=='Acute sectio') 
source('getTransferStats.R')
STAT_w36_T1m_csec_acute <- STAT
permSTAT_w36_T1m_csec_acute <- permSTATfisher

# w36 vs T month 3
phy2 <- phyX %>% subset_samples(Time == '3m' & Type == 'T' & DELIVERY=='Acute sectio') 
source('getTransferStats.R')
STAT_w36_T3m_csec_acute <- STAT
permSTAT_w36_T3m_csec_acute <- permSTATfisher

# save.image('./tmp_backup.RData')
save.image('./ORresults.RData')
```

## ALL - Compare weighted Ratios between c-section and vaginal birth at each timepoint

A permutation test between c-sectio and vaginal birth are conducted for all combinations. 

```{r}
nperm <-1000
# w36 vs F week 1
phy1 <- phyX %>% subset_samples(Type=='V' & Time == '36')
phy2 <- phyX %>% subset_samples(Time == '1w' & Type == 'F') 
source('inferenceTransferStat.R')
WeigtedRatio_F1w <- wrperm

phy2 <- phyX %>% subset_samples(Time == '1m' & Type == 'F') 
source('inferenceTransferStat.R')
WeigtedRatio_F1m <- wrperm


phy2 <- phyX %>% subset_samples(Time == '1y' & Type == 'F') 
source('inferenceTransferStat.R')
WeigtedRatio_F1y <- wrperm

phy2 <- phyX %>% subset_samples(Time == '1w' & Type == 'T') 
source('inferenceTransferStat.R')
WeigtedRatio_T1w <- wrperm

phy2 <- phyX %>% subset_samples(Time == '1m' & Type == 'T') 
source('inferenceTransferStat.R')
WeigtedRatio_T1m <- wrperm

phy2 <- phyX %>% subset_samples(Time == '3m' & Type == 'T') 
source('inferenceTransferStat.R')
WeigtedRatio_T3m <- wrperm

getPermutationPV <- function(WR){
  # WR <- WeigtedRatio_F1m
  pv <- sum(WR$Perm_ratioratio>WR$Model_ratioratio[1]) / dim(WR)[1]
  df <- data.frame(Model_ratioratio = WR$Model_ratioratio[1],
                   niter = dim(WR)[1]/2, 
                   Perm_ratioratio_median = median(WR$Perm_ratioratio), 
                   Perm_ratioratio_mean = mean(WR$Perm_ratioratio),
                   pv = pv)
  return(df)
}
WRpermstats <- rbind(
  data.frame(Type = 'Fecal', Time = 7, getPermutationPV(WeigtedRatio_F1w)),
  data.frame(Type = 'Fecal', Time = 30, getPermutationPV(WeigtedRatio_F1m)),
  data.frame(Type = 'Fecal', Time = 300, getPermutationPV(WeigtedRatio_F1y)),
  data.frame(Type = 'Airways', Time = 7, getPermutationPV(WeigtedRatio_T1w)),
  data.frame(Type = 'Airways', Time = 30, getPermutationPV(WeigtedRatio_T1m)),
  data.frame(Type = 'Airways', Time = 90, getPermutationPV(WeigtedRatio_T3m)))


# save.image('./weighted_permutation_results.RData')
save(file = './weighted_permutation_results_onesided.RData', list = c('WRpermstats'))
```


```{r, eval=TRUE}
load('./weighted_permutation_results_onesided.RData')

# load('./tmp_backup.RData')
load('./ORresults.RData')
STATtot <- rbind(
  data.frame(STAT_w36_F1m_csec, time =  30, type =  'Fecal', delivery =  'csec'),
  data.frame(STAT_w36_F1m_norm, time =  30, type =  'Fecal', delivery =  'norm'),
  data.frame(STAT_w36_F1w_csec, time =  7, type =  'Fecal', delivery =  'csec'),
  data.frame(STAT_w36_F1w_norm, time =  7, type =  'Fecal', delivery =  'norm'),
  data.frame(STAT_w36_F1y_csec, time =  300, type =  'Fecal', delivery =  'csec'),
  data.frame(STAT_w36_F1y_norm, time =  300, type =  'Fecal', delivery =  'norm'),
  data.frame(STAT_w36_T1m_csec, time =  30, type =  'Airways', delivery =  'csec'),
  data.frame(STAT_w36_T1m_norm, time =  30, type =  'Airways', delivery =  'norm'),
  data.frame(STAT_w36_T1w_csec, time =  7, type =  'Airways', delivery =  'csec'),
  data.frame(STAT_w36_T1w_norm, time =  7, type =  'Airways', delivery =  'norm'),
  data.frame(STAT_w36_T3m_csec, time =  90, type =  'Airways', delivery =  'csec'),
  data.frame(STAT_w36_T3m_norm, time =  90, type =  'Airways', delivery =  'norm'),
  data.frame(STAT_w36_F1m_csec_planned, time = 30, type = 'Fecal', delivery = 'csec_planned'),
  data.frame(STAT_w36_F1w_csec_planned, time = 7, type = 'Fecal', delivery = 'csec_planned'),
  data.frame(STAT_w36_F1y_csec_planned, time = 300, type = 'Fecal', delivery = 'csec_planned'),
  data.frame(STAT_w36_T1w_csec_planned, time = 7, type = 'Airways', delivery = 'csec_planned'),
  data.frame(STAT_w36_T1m_csec_planned, time = 30, type = 'Airways', delivery = 'csec_planned'),
  data.frame(STAT_w36_T3m_csec_planned, time = 90, type = 'Airways', delivery = 'csec_planned'),
  data.frame(STAT_w36_F1m_csec_acute, time = 30, type = 'Fecal', delivery = 'csec_acute'),
  data.frame(STAT_w36_F1w_csec_acute, time = 7, type = 'Fecal', delivery = 'csec_acute'),
  data.frame(STAT_w36_F1y_csec_acute, time = 300, type = 'Fecal', delivery = 'csec_acute'),
  data.frame(STAT_w36_T1w_csec_acute, time = 7, type = 'Airways', delivery = 'csec_acute'),
  data.frame(STAT_w36_T1m_csec_acute, time = 30, type = 'Airways', delivery = 'csec_acute'),
  data.frame(STAT_w36_T3m_csec_acute, time = 90, type = 'Airways', delivery = 'csec_acute'))

n_fam <- 15

d <- STATtot %>% 
  filter(time==7 & type == 'Fecal') %>%
  group_by(Family) %>%
  summarise(abu = sum(abuMrel)) 
d <- d[order(d$abu,decreasing = T),]
STATtot$Family2 <-STATtot$Family <- STATtot$Family %>% as.character()

# set legend and colors
Fam_keep <- d$Family[1:(n_fam-1)] %>% as.character()
STATtot$Family2[!(STATtot$Family2 %in% Fam_keep)] <- 'other'

# table(STATtot$Family2)
legend_names <- as.character(levels(factor(STATtot$Family2)))

cols  <- c(brewer.pal(8,"Set1"), brewer.pal(7,"Dark2"),brewer.pal(7,"Set2"),brewer.pal(12,"Set3"),brewer.pal(7,"Accent"),brewer.pal(12,"Paired"),"gray") 
cols <- c(cols[1:(length(legend_names)-1)],'gray')
```

## ALL most abundant models (Descriptives and inference)

```{r}
library(doMC)
nperm <- 1000
# w36 
phy1 <- phyX %>% subset_samples(Type=='V' & Time == '36')
sd1 <- sample_data(phy1) %>% mutate(delivery = DELIVERY, delivery = replace(delivery, delivery!='Normal','Sectio'))
ph <- filter_taxa(phy1,function(x) sum(x>0)>0, TRUE)
X1 <- data.frame(sd1,t(otu_table(ph)))

# vs F week 1
phy2 <- phyX %>% subset_samples(Time == '1w' & Type == 'F')
source('getWinnerStats.R')
winner_w36_F1w <- tb

# vs F week 1
phy2 <- phyX %>% subset_samples(Time == '1w' & Type == 'F')
source('getWinnerStats.R')
winner_w36_F1w <- tb


# vs F month 1
phy2 <- phyX %>% subset_samples(Time == '1m' & Type == 'F')
source('getWinnerStats.R')
winner_w36_F1m <- tb

# vs F year 1
phy2 <- phyX %>% subset_samples(Time == '1y' & Type == 'F')
source('getWinnerStats.R')
winner_w36_F1y <- tb

# vs T week 1
phy2 <- phyX %>% subset_samples(Time == '1w' & Type == 'T')
source('getWinnerStats.R')
winner_w36_T1w <- tb


# vs T month 1
phy2 <- phyX %>% subset_samples(Time == '1m' & Type == 'T')
source('getWinnerStats.R')
winner_w36_T1m <- tb

# vs T month 3
phy2 <- phyX %>% subset_samples(Time == '3m' & Type == 'T')
source('getWinnerStats.R')
winner_w36_T3m <- tb

## Descriptives

# w36 
phy1 <- phyX %>% subset_samples(Type=='V' & Time == '36')
sd1 <- sample_data(phy1) %>% mutate(delivery = DELIVERY, delivery = replace(delivery, delivery!='Normal','Sectio'))
ph <- filter_taxa(phy1,function(x) sum(x>0)>0, TRUE)
X1 <- data.frame(sd1,t(otu_table(ph)))

MostAbundantStats <- rbind(
  data.frame(winner_w36_F1w,Time = 7,Type = 'Fecal'),
  data.frame(winner_w36_F1m,Time = 30,Type = 'Fecal'),
  data.frame(winner_w36_F1y,Time = 300,Type = 'Fecal'),
  data.frame(winner_w36_T1w,Time = 7,Type = 'Airways'),
  data.frame(winner_w36_T1m,Time = 30,Type = 'Airways'),
  data.frame(winner_w36_T3m,Time = 90,Type = 'Airways'))

# save.image('./Winnerstats.RData')
save(file = './Winnerstats.RData', list = c('MostAbundantStats','X1'))
```


In this analysis, the vaginal dominating ASV in each mother is looked for in the corresponding child. This analysis is ASV _unspecific_ I.e. just the dominating one we look for. The figure below shows the frequency on the y-axis of the ASV in the child, color indicate delivery mode, x-axis the domination rank (1 = most dominating, 2 = second,...), label refers to p-values towards H0 of no transfer. 

```{r,eval = TRUE}
load('./Winnerstats.RData')

g1 <- ggplot(data = MostAbundantStats,aes(rnk,prcModel*100 , color = delivery, label = pv)) + 
  geom_point(aes(size = -log10(pv + 0.0001)))  + 
  geom_line() +
  geom_text(data = MostAbundantStats[MostAbundantStats$pv<1.05,],color = 'black', size = 3) + 
  facet_wrap(~factor(Type):factor(Time)) + 
  xlab('Most Abundant in Mothers Rank') + 
  ylab('Percent observed in child') + 
  theme_bw() + 
  theme(legend.position = 'bottom')

print(g1)
# txtb <- phyX %>% tax_table() %>% data.frame

mX1 <- X1 %>% 
  gather(ASV,Mcount,-c(dyadnb,Time,Type,DELIVERY,delivery)) %>%
  group_by(dyadnb) %>%
  arrange(desc(Mcount)) %>%
  mutate(rnk = 1:n()) %>%
  filter(rnk<20) %>%
  group_by(rnk,ASV) %>%
  summarise(n = n()) %>%
  ungroup %>%
  group_by(rnk) %>%
  arrange(desc(n)) %>%
  mutate(ordr = 1:n()) %>%
  filter(ordr<10 & n>10) %>%
  ungroup %>%
  arrange(rnk)

mX1 %>% 
  filter(rnk<5) %>%
  data.frame() %>%
  kable(caption = 'Ranking of ASVs. I.e. which ASVs are dominating at which rank and in how many children')
```

### Table 1 - testable ASV's 

```{r,eval=TRUE}
getFDR <- function(x){
  # x <- STATtot
  nASV = dim(x)[1] 
  relativeAbuM <- 100*sum(x$abuMrel)
  relativeAbuC <-  100*sum(x$abuCrel)
  pv <- x$Fisher_p.value
  pvadj <- p.adjust(pv, 'fdr')
  pmin = min(pv)
  pminadj = min(pvadj)
  n_crude_below_01 = sum(pv<=0.01)
  n_crude_below_05 = sum(pv<=0.05)
  n_fdr_below_10 = sum(pvadj<=0.1)
  n_fdr_below_05= sum(pvadj<=0.05)
  df <- data.frame(nASV,relativeAbuM,relativeAbuC,
                   pmin,pminadj,
                   n_crude_below_01,n_crude_below_05,
                   n_fdr_below_05,n_fdr_below_10)
  df
}
ttb <- STATtot %>%
  filter(delivery %in% c('csec','norm')) %>%
  group_by(delivery, time,type) %>%
  # summarise()
  do(getFDR(x = .))

kable(ttb,caption = 'Individual transfermodels, coverage of testable ASVs and strongest results', digits = 3)
rio::export(ttb,file = 'Table1.xlsx')
```

### Figure S3 - vaginal delivery

The odds for transfer between mother (week 36), child (week 1). Top panel shows the OR (x-axis) and the strength (p-value). Lower panel shows OR (y-axis) versus the population wide vaginal abundance (x-axis). This shows, that 1) there is trend of transfer from more ASV's being positive (OR>1) than negative, more signal in fecal, that none is strongly significant controling for fdr, and that those which obtain the strongest tranfer results are those which are in low populationwide vaginal abundance. 

```{r,eval=TRUE}
g_legend<-function(a.gplot){ 
  tmp <- ggplot_gtable(ggplot_build(a.gplot)) 
  leg <- which(sapply(tmp$grobs, function(x) x$name) == "guide-box") 
  legend <- tmp$grobs[[leg]] 
  return(legend)} 
mytheme <- theme(axis.title = element_text(size = 15), axis.text = element_text(size = 10)) + theme_bw()

DF <- STATtot %>%
  filter(time==7 & delivery=='norm')

n_labels <- 30
DFlab <- DF %>%
  filter(Fisher_p.value<0.05) %>%
  arrange(round(log10(Fisher_p.value)), -abuC) 
nn <- dim(DFlab)[1]

g1 <- 
  ggplot(data = DF,aes(x = Fisher_estimatetr,y = -log10(Fisher_p.value),color = Family2,label = ASV,size = abuCrel))+ 
  geom_point() + 
  facet_wrap(~type) + 
  geom_text_repel(color = 'black',fill = NA,data = DFlab[1:min(nn,n_labels),],size = 3) + 
  geom_hline(yintercept = -log10(0.05)) + theme_bw() +
  scale_size(guide = "none") +
  scale_color_manual(values = cols,labels = legend_names) + 
  geom_vline(xintercept = 1) + 
  scale_x_log10() +
  labs(col = '')+
  guides(colour = guide_legend(override.aes = list(size=5))) + 
  theme(legend.position = 'bottom') + xlab('Odds ratio')

# Set data
df <- DF
# Calculate trend line for type
tb <- df %>%
  group_by(type) %>%
  do(lm(data = ., log10(Fisher_estimatetr)~log10(abuMrel),weights = -log10(Fisher_p.value)) %>% tidy) %>%
  select(term,estimate) %>%
  spread(term,estimate) %>%
  rename(a = `(Intercept)`, 
         b = `log10(abuMrel)`)

g11 <- ggplot(data = df,
              aes(x = abuMrel,y =  Fisher_estimatetr,color = Family2, label = ASV)) +
  geom_point(size = 3) +
  facet_wrap(~type)  + 
  scale_x_log10() + 
  scale_y_log10() + 
  geom_abline(data = tb, aes(intercept = a, slope = b)) + 
  geom_hline(yintercept = 1) + 
  scale_color_manual(values = cols,labels = legend_names)  + xlab('Population Abundance Vaginal') + ylab('Odds ratio')

legend <- g_legend(g1) 

G1 <- ggdraw() + draw_plot(g1 +  mytheme + theme(legend.position = "hidden"), 0, .55, 1, .45) + 
  draw_plot(g11 + mytheme+ theme(legend.position = "hidden"), 0, 0.1, 1, .45)+
  draw_grob(legend, 0, 0, 1, .1) 
print(G1)

pdf('FigureS3.pdf', height = 13, width = 13)
G1
dev.off()

tb <- STATtot %>%
  # filter(time==7) %>%
  filter(delivery %in% c('norm','csec')) %>%
  group_by(type,time,delivery) %>%
  do(lm(data = ., log10(Fisher_estimatetr)~log10(abuMrel),weights = -log10(Fisher_p.value)) %>% tidy) %>%
  filter(term!='(Intercept)') %>%
  select(-statistic)

kable(tb, digits = 4,caption = 'inference for relation between odds for tranfers and population maternal abundance')
```


### Figure S4 - sectio delivery

Same figure, just for sectio born children, where the signal is diluted. In this analysis the transfer to fecal is strongly inversely associated with population wide vaginal abundance.

```{r, eval=TRUE}
# Number of testable ASVs: "found 68 and 103 ASVs obtained a dynamic range for transfer to hypopharyngeal and fecal samples"
# Vulcano no dico-odds for tranfer
# Statistics on proportion of positive odds:  "factor of 1.8 (1.3-2.6) and 3.9 (2.8-5.4) (p < 0.001 for both) for hypopharyngeal and fecal samples"

DF <- STATtot %>%
  filter(time==7 & delivery=='csec_acute')

DFlab <- DF %>%
  filter(Fisher_p.value<0.05) %>%
  arrange(round(log10(Fisher_p.value)), -abuC) 
nn <- dim(DFlab)[1]

g1 <- 
  ggplot(data = DF,aes(x = Fisher_estimatetr,y = -log10(Fisher_p.value),color = Family2,label = ASV,size = abuCrel))+ 
  geom_point() + 
  facet_wrap(~type) + 
  geom_text_repel(color = 'black',fill = NA,data = DFlab[1:min(nn,n_labels),],size = 3) + 
  geom_hline(yintercept = -log10(0.05)) + theme_bw() +
  scale_size(guide = "none") +
  scale_color_manual(values = cols,labels = legend_names) + 
  geom_vline(xintercept = 1) + 
  scale_x_log10() +
  labs(col = '')+
  guides(colour = guide_legend(override.aes = list(size=5))) + 
  theme(legend.position = 'bottom') + xlab('Odds ratio')


# Set data
df <- DF
# Calculate trend line for type
tb <- df %>%
  group_by(type) %>%
  do(lm(data = ., log10(Fisher_estimatetr)~log10(abuMrel),weights = -log10(Fisher_p.value)) %>% tidy) %>%
  select(term,estimate) %>%
  spread(term,estimate) %>%
  rename(a = `(Intercept)`, 
         b = `log10(abuMrel)`)

g11 <- ggplot(data = df,
              aes(x = abuMrel,y =  Fisher_estimatetr,color = Family2, label = ASV)) +
  geom_point(size = 3) +
  facet_wrap(~type)  + 
  scale_x_log10() + 
  scale_y_log10() + 
  geom_abline(data = tb, aes(intercept = a, slope = b)) + 
  geom_hline(yintercept = 1) + 
  scale_color_manual(values = cols,labels = legend_names)  + xlab('Population Abundance Vaginal') + ylab('Odds ratio')

legend <- g_legend(g1) 

G2 <- ggdraw() + draw_plot(g1 +  mytheme + theme(legend.position = "hidden"), 0, .55, 1, .45) + 
  draw_plot(g11 + mytheme+ theme(legend.position = "hidden"), 0, 0.1, 1, .45)+
  draw_grob(legend, 0, 0, 1, .1) 
print(G2)

pdf('FigureS4.pdf', height = 13, width = 13)
G2
dev.off()


```


## Weighted Odds Ratio

In order to make a commen measure for the tranfer signal, a weigthed ratio of positive vs negative ASV's. For each ASV the ratio is defined as the sum of -log(OR)*log(p_value) which should be around 1 in case of no tranfer, and larger when present. To test for transfer, the dyads are scrampled to construct a null distribution for the ratio. 

Due to the high sparsity, the null distribution is not always centered around 1, therefore the ratios reported is the model ratio relative to the median of the null distribution. 


### OVERALL ratio between positive and negative odds

The figure shows time point on x-axis, ratio on y-axis color is mode of delivery and panel is compartment. The text reflects the p-value. 


```{r}
tbWeigtedSTATs <- STATtot %>%
  mutate(Fisher_estimatetr = truncateZerosInf(Fisher_estimate)) %>%
  filter(delivery %in% c('csec','norm')) %>%
  group_by(type,delivery,time) %>%
  do(getWeigtedRatio(x = .)) 

permstats <- rbind(
  extractPV(permSTAT_w36_F1w_csec, modelratio = tbWeigtedSTATs$ratio[tbWeigtedSTATs$time==7 & tbWeigtedSTATs$type=='Fecal' & tbWeigtedSTATs$delivery=='csec']), 
  extractPV(permSTAT_w36_F1m_csec, modelratio = tbWeigtedSTATs$ratio[tbWeigtedSTATs$time==30 & tbWeigtedSTATs$type=='Fecal' & tbWeigtedSTATs$delivery=='csec']), 
  extractPV(permSTAT_w36_F1y_csec, modelratio = tbWeigtedSTATs$ratio[tbWeigtedSTATs$time==300 & tbWeigtedSTATs$type=='Fecal' & tbWeigtedSTATs$delivery=='csec']), 
  extractPV(permSTAT_w36_F1w_norm, modelratio = tbWeigtedSTATs$ratio[tbWeigtedSTATs$time==7 & tbWeigtedSTATs$type=='Fecal' & tbWeigtedSTATs$delivery=='norm']), 
  extractPV(permSTAT_w36_F1m_norm, modelratio = tbWeigtedSTATs$ratio[tbWeigtedSTATs$time==30 & tbWeigtedSTATs$type=='Fecal' & tbWeigtedSTATs$delivery=='norm']), 
  extractPV(permSTAT_w36_F1y_norm, modelratio = tbWeigtedSTATs$ratio[tbWeigtedSTATs$time==300 & tbWeigtedSTATs$type=='Fecal' & tbWeigtedSTATs$delivery=='norm']), 
  extractPV(permSTAT_w36_T1w_csec, modelratio = tbWeigtedSTATs$ratio[tbWeigtedSTATs$time==7 & tbWeigtedSTATs$type=='Airways' & tbWeigtedSTATs$delivery=='csec']), 
  extractPV(permSTAT_w36_T1m_csec, modelratio = tbWeigtedSTATs$ratio[tbWeigtedSTATs$time==30 & tbWeigtedSTATs$type=='Airways' & tbWeigtedSTATs$delivery=='csec']), 
  extractPV(permSTAT_w36_T3m_csec, modelratio = tbWeigtedSTATs$ratio[tbWeigtedSTATs$time==90 & tbWeigtedSTATs$type=='Airways' & tbWeigtedSTATs$delivery=='csec']),
  extractPV(permSTAT_w36_T1w_norm, modelratio = tbWeigtedSTATs$ratio[tbWeigtedSTATs$time==7 & tbWeigtedSTATs$type=='Airways' & tbWeigtedSTATs$delivery=='norm']), 
  extractPV(permSTAT_w36_T1m_norm, modelratio = tbWeigtedSTATs$ratio[tbWeigtedSTATs$time==30 & tbWeigtedSTATs$type=='Airways' & tbWeigtedSTATs$delivery=='norm']), 
  extractPV(permSTAT_w36_T3m_norm, modelratio = tbWeigtedSTATs$ratio[tbWeigtedSTATs$time==90 & tbWeigtedSTATs$type=='Airways' & tbWeigtedSTATs$delivery=='norm']) 
) 

# tbWeigtedSTATs$p_value <- p.values
tbWeigtedSTATs <- cbind(data.frame(tbWeigtedSTATs),permstats)

save(file = './RatioStats_onesided.RData',list = c('tbWeigtedSTATs'))

```

### Weigted Transfer Ratios

```{r,eval = TRUE}
load('./RatioStats_onesided.RData')
tbWeigtedSTATs <- tbWeigtedSTATs %>%
  mutate(model_over_perm = modelratio/permmedian)

tb1 <- kable(tbWeigtedSTATs, caption = 'Weigthed Transfer Odds as function of delivery mode, compartment and age',digits = 5)
tb1

```

### Figure 1 - Weigted Transfer Ratios

```{r,eval = TRUE, fig.height=5, fig.width=10}
g1 <- ggplot(data = tbWeigtedSTATs, 
             aes(time,modelratio,  
                 ymin = exp(log(modelratio) - SElgratio),
                 ymax = exp(log(modelratio) + SElgratio),
                 color = delivery, group = delivery:type, label = paste('p =',pv))) + 
  geom_line(size = 1, position = position_dodge(width = 0.1)) + 
  geom_point(size = 2,position = position_dodge(width = 0.1)) + 
  #geom_text(color = 'black') + 
  scale_x_log10() + 
  scale_color_manual(values = c('red','blue')) + 
  # scale_y_log10() + 
  geom_hline(yintercept = 1) + 
  geom_errorbar(width = 0.1,position = position_dodge(width = 0.1)) + 
  # scale_color_manual(values = c('brown','blue','steelblue','darkblue') ) + 
  xlab('Age (Days)') + 
  facet_wrap(~type) +
  theme_bw() + 
  theme(legend.position = 'none',panel.grid.minor = element_blank())

pdf('Figure1.pdf',width = 5,height = 2.5)
g1 + ylab('WTR')
dev.off()

kable(WRpermstats, digits = 2, caption = 'Inference for the difference in weigted ratios between sectio- and vaginal born children')

```

### Calculated for individual taxonomic levels

Mostly based on family, but for families with low number of testable ASV's these are merged based on the Phylum level.

Same type of figure as above. 

```{r}
# merge on large families, otherwise on phylum level
tbsel <- STATtot %>%
  filter(delivery %in% c('csec','norm')) %>%
  group_by(Family,time,type,delivery) %>%
  summarise(n = n()) %>%
  ungroup() %>%
  group_by(Family) %>%
  summarise(nmin = min(n),
            nmax = max(n)) %>%
  ungroup() %>%
  left_join(TAXtb[!duplicated(TAXtb$Family    ),1:5], by = c('Family'= 'Family'))

STATtot$Family3 <- STATtot$Family %>% as.character()
STATtot$Family3[STATtot$Family %in% tbsel$Family[tbsel$nmin<5]] <- STATtot$Phylum[STATtot$Family %in% tbsel$Family[tbsel$nmin<5]] %>% as.character()
ttb <- table(STATtot$Family3)
ttb <- ttb[order(ttb,decreasing = T)]

STATtot$Family3[STATtot$Family3 %in% names(ttb[10:length(ttb)])] <- 'other'

tbsel <- STATtot %>%
  filter(delivery %in% c('csec','norm')) %>%
  group_by(Family3,time,type,delivery) %>%
  summarise(n = n()) %>%
  ungroup() %>%
  group_by(Family3) %>%
  summarise(nmin = min(n),
            nmax = max(n), 
            nmedian = median(n)) %>%
  ungroup() 


tb2 <- STATtot %>%
  filter(delivery %in% c('csec','norm')) %>%
  group_by(Family3,time,type,delivery) %>%
  mutate(nt = n()) %>%
  # filter(nt>5) %>%
  do(getWeigtedRatio(x = .)) %>%
  filter(!is.na(ratio) & !is.infinite(ratio)) 

# get the corresponding PVs and median perm ratio
permSTATall_df <- data.frame(type = c(rep('Fecal',6),rep('Airways',6)), time = c(7,30,300,7,30,300,7,30,90,7,30,90), delivery =as.character(c( rep('csec',3),rep('norm',3))))

permSTATall <- list(permSTAT_w36_F1w_csec, 
                    permSTAT_w36_F1m_csec, 
                    permSTAT_w36_F1y_csec, 
                    permSTAT_w36_F1w_norm, 
                    permSTAT_w36_F1m_norm, 
                    permSTAT_w36_F1y_norm, 
                    permSTAT_w36_T1w_csec, 
                    permSTAT_w36_T1m_csec, 
                    permSTAT_w36_T3m_csec, 
                    permSTAT_w36_T1w_norm, 
                    permSTAT_w36_T1m_norm, 
                    permSTAT_w36_T3m_norm)

# permstat_df <- c()
# for (i in 1:dim(tb2)[1]){
for (i in 8:dim(tb2)[1]){
  # for (i in 1:2){
  # i <- 1
  print(i)
  rw <- tb2[i,]
  print(rw)
  ASVsel <- unique(STATtot$ASV[STATtot$Family3==rw$Family3])
  idL <- permSTATall_df$type==rw$type & permSTATall_df$time==rw$time & permSTATall_df$delivery==as.character(rw$delivery) 
  permSTAT <- permSTATall[[which(idL)]]
  permSTAT <- permSTAT[dimnames(permSTAT)[[1]] %in% ASVsel,,]
  permstat_df <- rbind(permstat_df,data.frame(extractPV(permSTAT,modelratio = rw$ratio),nASVs = dim(permSTAT)[1]))
  # print(dim(permSTAT))
}

FamilyWRstats <- cbind(data.frame(tb2),permstat_df) 

save(file = './FamilyRatioSTATs.RData',list = c('FamilyWRstats','tbsel','STATtot'))
```



```{r,eval = TRUE, fig.height=20, fig.width=5, fig.align='center'}
load('./FamilyRatioSTATs.RData')

kable(tbsel, caption = 'Number of ASVs for each taxonomic partitioning based on Family or Phylum across all models')

g3 <- ggplot(data = FamilyWRstats, aes(time,modelratio/permmedian, 
                                       color = delivery, group = delivery:type, 
                                       linetype = type, 
                                       label = pv)) + 
  geom_line(size = 1) + 
  geom_point(aes(size = log10(nn +np))) + 
  geom_text(color = 'black') + 
  # facet_wrap(~Family3,ncol = 3) + 
  facet_grid(Family3~type) + 
  # scale_color_manual(values = c('brown','blue') ) + 
  scale_x_log10() + 
  # scale_y_log10() + 
  scale_y_log10(breaks=c(0.25,0.5,1,2,4,8,16),labels=c(0.25,0.5,1,2,4,8,16)) + 
  # scale_y_sqrt() + 
  geom_hline(yintercept = 1) + 
  xlab('Age (Days)') + 
  theme_bw() + 
  theme(legend.position = 'top')

g3
pdf('FigureS4.pdf',height = 20, width = 5)
g3
dev.off()
```


```{r}
# merge on large families. 
tbsel <- STATtot %>%
  filter(delivery %in% c('csec','norm')) %>%
  group_by(Family,time,type,delivery) %>%
  summarise(n = n()) %>%
  ungroup() %>%
  group_by(Family) %>%
  summarise(nmin = min(n),
            nmax = max(n)) %>%
  ungroup() %>%
  left_join(TAXtb[!duplicated(TAXtb$Family    ),1:5], by = c('Family'= 'Family')) %>%
  arrange(desc(nmin)) %>%
  filter((nmin>6 | nmax > 15) & nmin>2)

STATtot$Family3 <- STATtot$Family %>% as.character()
STATtot2 <- STATtot %>% filter(Family3 %in% tbsel$Family)
ttb <- table(STATtot2$Family3)
ttb <- ttb[order(ttb,decreasing = T)]

tbsel <- STATtot2 %>%
  filter(delivery %in% c('csec','norm')) %>%
  group_by(Family3,time,type,delivery) %>%
  summarise(n = n()) %>%
  ungroup() %>%
  group_by(Family3) %>%
  summarise(nmin = min(n),
            nmax = max(n), 
            nmedian = median(n)) %>%
  ungroup() 


tb2 <- STATtot2 %>%
  filter(delivery %in% c('csec','norm')) %>%
  group_by(Family3,time,type,delivery) %>%
  mutate(nt = n()) %>%
  # filter(nt>5) %>%
  do(getWeigtedRatio(x = .)) %>%
  filter(!is.na(ratio) & !is.infinite(ratio)) 

# get the corresponding PVs and median perm ratio
permSTATall_df2 <- data.frame(type = c(rep('Fecal',6),rep('Airways',6)), time = c(7,30,300,7,30,300,7,30,90,7,30,90), delivery =as.character(c( rep('csec',3),rep('norm',3))))

permSTATall <- list(permSTAT_w36_F1w_csec, 
                    permSTAT_w36_F1m_csec, 
                    permSTAT_w36_F1y_csec, 
                    permSTAT_w36_F1w_norm, 
                    permSTAT_w36_F1m_norm, 
                    permSTAT_w36_F1y_norm, 
                    permSTAT_w36_T1w_csec, 
                    permSTAT_w36_T1m_csec, 
                    permSTAT_w36_T3m_csec, 
                    permSTAT_w36_T1w_norm, 
                    permSTAT_w36_T1m_norm, 
                    permSTAT_w36_T3m_norm)

permstat_df2 <- c()
for (i in 1:dim(tb2)[1]){
  # for (i in 1:2){
  # i <- 1
  print(i)
  rw <- tb2[i,]
  ASVsel <- unique(STATtot2$ASV[STATtot2$Family3==rw$Family3])
  idL <- permSTATall_df2$type==rw$type & permSTATall_df2$time==rw$time & permSTATall_df2$delivery==as.character(rw$delivery) 
  permSTAT <- permSTATall[[which(idL)]]
  permSTAT <- permSTAT[dimnames(permSTAT)[[1]] %in% ASVsel,,]
  permstat_df2 <- rbind(permstat_df2,data.frame(extractPV(permSTAT,modelratio = rw$ratio),nASVs = dim(permSTAT)[1]))
  # print(dim(permSTAT))
}

FamilyWRstats2 <- cbind(data.frame(tb2),permstat_df2) 

save(file = './FamilyRatioSTATs_v2.RData',list = c('FamilyWRstats2','tbsel','STATtot2'))
```

### On a higher level than Family

```{r}
# merge on large order. 
tbsel <- STATtot %>%
  filter(delivery %in% c('csec','norm')) %>%
  group_by(Order,time,type,delivery) %>%
  summarise(n = n()) %>%
  ungroup() %>%
  group_by(Order) %>%
  summarise(nmin = min(n),
            nmax = max(n)) %>%
  ungroup() %>%
  left_join(TAXtb[!duplicated(TAXtb$Order      ),1:4], by = c('Order'= 'Order')) %>%
  arrange(desc(nmax)) %>%
  filter(nmax > 10)

STATtot2 <- STATtot %>% 
  mutate(Order2 = Order %>% as.character()) %>% 
  filter(Order2 %in% tbsel$Order) 

ttb <- table(STATtot2$Order2)
ttb <- ttb[order(ttb,decreasing = T)]

tbsel <- STATtot2 %>%
  filter(delivery %in% c('csec','norm')) %>%
  group_by(Order2,time,type,delivery) %>%
  summarise(n = n()) %>%
  ungroup() %>%
  group_by(Order2) %>%
  summarise(nmin = min(n),
            nmax = max(n), 
            nmedian = median(n)) %>%
  ungroup() 


tb2 <- STATtot2 %>%
  filter(delivery %in% c('csec','norm')) %>%
  group_by(Order2,time,type,delivery) %>%
  mutate(nt = n()) %>%
  # filter(nt>5) %>%
  do(getWeigtedRatio(x = .)) %>%
  filter(!is.na(ratio) & !is.infinite(ratio)) 

# get the corresponding PVs and median perm ratio
permSTATall_df3 <- data.frame(type = c(rep('Fecal',6),rep('Airways',6)), time = c(7,30,300,7,30,300,7,30,90,7,30,90), delivery =as.character(c( rep('csec',3),rep('norm',3))))

permSTATall <- list(permSTAT_w36_F1w_csec, 
                    permSTAT_w36_F1m_csec, 
                    permSTAT_w36_F1y_csec, 
                    permSTAT_w36_F1w_norm, 
                    permSTAT_w36_F1m_norm, 
                    permSTAT_w36_F1y_norm, 
                    permSTAT_w36_T1w_csec, 
                    permSTAT_w36_T1m_csec, 
                    permSTAT_w36_T3m_csec, 
                    permSTAT_w36_T1w_norm, 
                    permSTAT_w36_T1m_norm, 
                    permSTAT_w36_T3m_norm)

permstat_df3 <- c()
for (i in 1:dim(tb2)[1]){
  # for (i in 1:2){
  # i <- 1
  print(i)
  rw <- tb2[i,]
  ASVsel <- unique(STATtot2$ASV[STATtot2$Order2==rw$Order2])
  idL <- permSTATall_df2$type==rw$type & permSTATall_df2$time==rw$time & permSTATall_df2$delivery==as.character(rw$delivery) 
  permSTAT <- permSTATall[[which(idL)]]
  permSTAT <- permSTAT[dimnames(permSTAT)[[1]] %in% ASVsel,,]
  permstat_df3 <- rbind(permstat_df3,data.frame(extractPV(permSTAT,modelratio = rw$ratio),nASVs = dim(permSTAT)[1]))
  #permstat_df3 <- rbind(permstat_df3,data.frame(pv = NA, SElgratio=NA, permmedian = NA, modelratio = NA,nASVs = dim(permSTAT)[1]))
  # print(dim(permSTAT))
}

OrderWRstats <- cbind(data.frame(tb2),permstat_df3) 
save(file = './OrderRatioSTATs.RData',list = c('OrderWRstats','tbsel','STATtot2'))
```


### Figure 2 - Individual taxonomic levels

```{r,eval = TRUE, fig.height=20, fig.width=5, fig.align='center'}
load('./FamilyRatioSTATs_v2.RData')

kable(tbsel, caption = 'Number of ASVs for each taxonomic partitioning based on Family or Phylum across all models')
FamilyWRstats2 <- FamilyWRstats2 %>%
  group_by(Family3) %>%
  mutate(nASVs_fam = mean(nASVs)) %>%
  ungroup %>%
  group_by(Family3, type) %>%
  mutate(nASVs_fam_type = mean(nASVs)) %>%
  ungroup 
  
FamilyWRstats2_sel <- FamilyWRstats2 %>%
  # filter(nASVs_fam>=30) %>%
  mutate(modelratio_tr = truncateZerosInf(modelratio,16), 
         Family4 = paste(str_pad(round(nASVs_fam),2,pad = '0'),Family3,sep = '_'), 
         Family4 = factor(Family4)) 

tbb <- FamilyWRstats2_sel %>%
  group_by(Family3) %>%
  summarise(N = mean(nASVs_fam)) %>%
  arrange(desc(N))

FamilyWRstats2_sel$Family4 = factor(FamilyWRstats2_sel$Family4, rev(levels(FamilyWRstats2_sel$Family4)))
FamilyWRstats2_sel$Family5 = factor(FamilyWRstats2_sel$Family3, tbb$Family3)

FamilyWRstats2_sel %>%
  select(-np,-nn,-ratio,-Family4) %>%
  rio::export(x = ., file = 'Family_WR.xlsx')

gg <- FamilyWRstats2 %>%
  # filter(nASVs_fam>=399) %>%
  ggplot(data = ., aes(modelratio,permmedian)) + geom_point()

plotWR <- function(df,brks = 2^c(-8:8)) {
  df %>% 
    group_by(Family3, time, type,delivery) %>% 
    mutate(ymax = min(exp(log(modelratio) + SElgratio),max(brks)),
           ymin = max(exp(log(modelratio) - SElgratio),min(brks))) %>%
  ggplot(data = ., aes(time,
                       # modelratio/permmedian, 
                       modelratio_tr,
                       ymax = ymax,
                       ymin = ymin,
                       color = delivery, 
                       group = delivery:type, 
                       #linetype = type, 
                       label = nASVs, 
                       linetype = nASVs_fam_type<=30)) + 
  geom_line(size = 1,position=position_dodge(width = 0.1)) + 
  #geom_point(aes(size = log10(nn +np))) + 
  geom_point(size = 2,position=position_dodge(width = 0.1)) + 
  geom_errorbar(width = 0.1, size = 1, position=position_dodge(width = 0.1)) + 
  scale_color_manual(values = c('red','blue')) + 
  # scale_alpha_discrete(breaks = c(0.1,1)) + 
  facet_grid(Family5~type) + 
  scale_x_log10() + 
  scale_y_log10(breaks=brks,labels=brks, limits = c(min(brks),max(brks))) + 
  geom_hline(yintercept = 1) + 
  xlab('Age (Days)') + 
  theme_bw() + 
  theme(legend.position = 'none',panel.grid.minor = element_blank())
}
brks <- 2^c(-4:4)
g3a1 <- FamilyWRstats2_sel %>%  filter(nASVs_fam>13) %>% plotWR(df = ., brks)
g3a2 <- FamilyWRstats2_sel %>%  filter(nASVs_fam<13) %>% plotWR(df = ., brks)

G3 <- ggdraw() + 
  draw_plot(g3a1 + ylab('WTR'),0,0,0.5,1) + 
  draw_plot(g3a2 + ylab(''),0.5,0,0.5,1) 

pdf('Figure2.pdf',height = 10, width = 10)
G3
dev.off()

G3n <- ggdraw() + 
  draw_plot(g3a1 + ylab('WR') + geom_text(color = 'black'),0,0,0.5,1) + 
  draw_plot(g3a2 + ylab('') + geom_text(color = 'black'),0.5,0,0.5,1) 

pdf('Figure2_withNs.pdf',height = 10, width = 10)
G3n
dev.off()


```



### Figure 2-Order - Individual taxonomic levels

```{r,eval = TRUE, fig.height=20, fig.width=5, fig.align='center'}
load('./OrderRatioSTATs.RData')

kable(tbsel, caption = 'Number of ASVs for each taxonomic partitioning based on Order across all models')
OrderWRstats <- OrderWRstats %>%
  group_by(Order2) %>%
  mutate(nASVs_fam = mean(nASVs)) %>%
  ungroup %>%
  group_by(Order2, type) %>%
  mutate(nASVs_order_type = mean(nASVs)) %>%
  ungroup %>% 
  ungroup
  
OrderWRstats_sel <- OrderWRstats %>%
  # filter(nASVs_fam>=30) %>%
  mutate(modelratio_tr = truncateZerosInf(modelratio,16), 
         Order3 = paste(str_pad(round(nASVs_order_type),2,pad = '0'),Order2,sep = '_'), 
         Order3 = factor(Order3)) 

tbb <- OrderWRstats_sel %>%
  group_by(Order2) %>%
  summarise(N = mean(nASVs_fam)) %>%
  arrange(desc(N))

OrderWRstats_sel$Order3 = factor(OrderWRstats_sel$Order3, rev(levels(OrderWRstats_sel$Order3)))
OrderWRstats_sel$Order4 = factor(OrderWRstats_sel$Order2, tbb$Order2)

OrderWRstats_sel %>%
  select(-np,-nn,-ratio,-Order3) %>%
  rio::export(x = ., file = 'Order_WR.xlsx')

gg <- OrderWRstats %>%
  # filter(nASVs_fam>=399) %>%
  ggplot(data = ., aes(modelratio,permmedian)) + geom_point()

plotWR <- function(df,brks = 2^c(-8:8)) {
  df %>% 
    group_by(Order2, time, type,delivery) %>% 
    mutate(ymax = min(exp(log(modelratio) + SElgratio),max(brks)),
           ymin = max(exp(log(modelratio) - SElgratio),min(brks))) %>%
  ggplot(data = ., aes(time,
                       # modelratio/permmedian, 
                       modelratio_tr,
                       ymax = ymax,
                       ymin = ymin,
                       color = delivery, 
                       group = delivery:type, 
                       #linetype = type, 
                       label = nASVs, 
                       linetype = nASVs_order_type<=30)) + 
  geom_line(size = 1,position=position_dodge(width = 0.1)) + 
  #geom_point(aes(size = log10(nn +np))) + 
  geom_point(size = 2,position=position_dodge(width = 0.1)) + 
  geom_errorbar(width = 0.1, size = 1, position=position_dodge(width = 0.1)) + 
  scale_color_manual(values = c('red','blue')) + 
  # scale_alpha_discrete(breaks = c(0.1,1)) + 
  facet_grid(Order4~type) + 
  scale_x_log10() + 
  scale_y_log10(breaks=brks,labels=brks, limits = c(min(brks),max(brks))) + 
  geom_hline(yintercept = 1) + 
  xlab('Age (Days)') + 
  theme_bw() + 
  theme(legend.position = 'none',panel.grid.minor = element_blank())
}
brks <- 2^c(-5:5)
g3a1 <- OrderWRstats_sel %>%  filter(Order2 %in% tbb$Order2[1:5]) %>% plotWR(df = ., brks)
g3a2 <- OrderWRstats_sel %>%  filter(Order2 %in% tbb$Order2[6:11]) %>% plotWR(df = ., brks)

G3 <- ggdraw() + 
  draw_plot(g3a1 + ylab('WTR'),0,0,0.5,1) + 
  draw_plot(g3a2 + ylab(''),0.5,0,0.5,1) 

pdf('Figure2_order.pdf',height = 10, width = 10)
G3
dev.off()

G3n <- ggdraw() + 
  draw_plot(g3a1 + ylab('WR') + geom_text(color = 'black'),0,0,0.5,1) + 
  draw_plot(g3a2 + ylab('') + geom_text(color = 'black'),0.5,0,0.5,1) 

pdf('Figure2_order_withNs.pdf',height = 10, width = 10)
G3n
dev.off()


```




### Figure S5 - Results on the tree of life

Here, we have the individual results shown on the phylogenetic tree (see _phylotree_transfer_rect.pdf_) 

```{r,eval=TRUE}
#############################
### PUT association results on a Phylo-tree
# select which of the ASV's that will be given a color 
# ic <- DF$p.value<0.9
# sum(ic)
# DFsel <- DF[ic,]
# AA <- reshape2::dcast(DFsel,ASV + Genus~comparison,value.var = 'estimate',mean)
# colnames(AA)[3:4] <- c('fecal','airways')

AA <- STATtot %>%
  filter(Fisher_p.value<0.9) %>%
  mutate(Fisher_estimatetr = truncateZerosInf(Fisher_estimate,10)) %>%
  mutate(cat = paste(type,time,delivery,sep = '_')) %>%
  select(ASV,Fisher_estimatetr,cat,Family3) %>%
  spread(cat,Fisher_estimatetr)

AA <- STATtot %>%
  filter(Fisher_p.value<0.9) %>%
  mutate(a = ceiling(sqrt(1-Fisher_p.value)*5)+1) %>%
  mutate(cat = paste(type,time,delivery,'a',sep = '_')) %>%
  select(ASV,a,cat,Family3) %>%
  spread(cat,a) %>%
  left_join(AA)

# ic1 <- rownames(tax_table(x123r)) %in% AA$ASV
# # truncate large estimates to 10 / 0.1
# ic <- abs(log10(AA$fecal))>1; ic[is.na(ic)] <- F
# AA$fecal[ic] <- 10^sign(log10(AA$fecal[ic]))
# ic <- abs(log10(AA$airways))>1; ic[is.na(ic)] <- F
# AA$airways[ic] <- 10^sign(log10(AA$airways[ic]))

# select which of the ASV's (in total) to include in the plotting
xASV <- otu_table(phyX)

ic1 <- rownames(xASV) %in% AA$ASV
ic2 <- apply(xASV>1,1,sum) > dim(xASV)[2]*0.1
ictaxa <- ic1 | ic2 

x <- subset_taxa(phyX,ictaxa)
# extract tree and taxonomic info
TREE <- phy_tree(x)
TXtab <- as.data.frame(tax_table(x))

# merge on inferential stats
AA <- merge(TXtab,AA,by.x = 'row.names',by.y = 'ASV',all = T)
size1 <- 0
size2 <- size1
# initiale tree
g3 <- ggtree(TREE,layout = 'slanted',branch.length="none")
g3 <- g3 %<+%  AA 
# change 'left side' labels
g3$data$label2 <- as.character(g3$data$Species)

# sel <- c('Fecal_7_norm',  'Fecal_30_norm',  'Fecal_300_norm',  'Fecal_7_csec',  'Fecal_30_csec',  'Fecal_300_csec',  'Airways_7_norm',  'Airways_30_norm',  'Airways_90_norm',  'Airways_7_csec',  'Airways_30_csec',  'Airways_90_csec')
dfg3 <- g3$data[g3$data$isTip,] %>%
  select(x,y,label) %>%
  left_join(STATtot, by = c('label' = 'ASV')) %>%
  filter(delivery %in% c('csec','norm')) %>%
  mutate(xtype = ifelse(type=='Fecal',0,8),
         xtime = ifelse(time==7,0,ifelse(time==30,1,2)), 
         xdelivery = ifelse(delivery=='norm',0,4), 
         xx = x + xtype + xtime + xdelivery+1, 
         z = -log10(Fisher_estimatetr)*log10(Fisher_p.value), 
         z = log(truncateZerosInf(10^z,10)))

# g3$data$label2 <- 
#   paste('ASV',unlist(lapply(as.list(g3$data$label2), function(x){strsplit(x,'_ASV')[[1]][2]})),'_') 
# ,unlist(lapply(as.list(g3$data$label2), function(x){strsplit(x,'_ASV')[[1]][1]})),sep = '')
g3 <- g3 + 
  #geom_tiplab(aes(x = x+2,label=Order,angle=angle,size =3)) +
  #geom_hilight(node=333, fill="darkgreen", alpha=.6) + 
  #geom_hilight(node=357, fill="gray10", alpha=.2) +
  #geom_hilight(node=348, fill="gray10", alpha=.2) +
  #geom_hilight(node=655, fill="gray10", alpha=.2) +
  #geom_hilight(node=336, fill="gray10", alpha=.2) +
  #geom_hilight(node=331, fill="gray10", alpha=.2) +
  #geom_hilight(node=476, fill="gray10", alpha=.2) +
  #geom_hilight(node=609, fill="gray10", alpha=.2) +
  # geom_tiplab(aes(x = x+15,label=Species,angle=angle,size =3, subset = ((angle<90 | angle>270) & isTip))) +
  # geom_tiplab(aes(x = x+15,label=label2,angle=angle+180,size =3, subset = ((angle>=90 & angle<=270) & isTip)), hjust =1) +
# geom_tippoint(aes(x = x+1, fill = log10(Fecal_7_norm),size = Fecal_7_norm_a + size1,subset = !is.na(Fecal_7_norm)), shape=22)  +
#   geom_tippoint(aes(x = x+2, fill = log10(Fecal_30_norm),size = Fecal_30_norm_a+ size1,subset = !is.na(Fecal_30_norm)), shape=22) +
#   geom_tippoint(aes(x = x+3, fill = log10(Fecal_300_norm),size = Fecal_300_norm_a+ size1,subset = !is.na(Fecal_300_norm)), shape=22) +
#   
#   geom_tippoint(aes(x = x+5, fill = log10(Fecal_7_csec),size = Fecal_7_csec_a+ size1,subset = !is.na(Fecal_7_csec)), shape=22)  +
#   geom_tippoint(aes(x = x+6, fill = log10(Fecal_30_csec),size = Fecal_30_csec_a+ size1,subset = !is.na(Fecal_30_csec)), shape=22) +
#   geom_tippoint(aes(x = x+7, fill = log10(Fecal_300_csec),size = Fecal_300_csec_a+ size1,subset = !is.na(Fecal_300_csec)), shape=22) +
#   
#   geom_tippoint(aes(x = x+9, fill = log10(Airways_7_norm),size = Airways_7_norm_a+size2,subset = !is.na(Airways_7_norm)), shape=22) +
#   geom_tippoint(aes(x = x+10, fill = log10(Airways_30_norm),size = Airways_30_norm_a+size2,subset = !is.na(Airways_30_norm)),shape=22) +
#   geom_tippoint(aes(x = x+11, fill = log10(Airways_90_norm),size = Airways_90_norm_a+size2,subset = !is.na(Airways_90_norm)),shape=22) +
#   
#   geom_tippoint(aes(x = x+13, fill = log10(Airways_7_csec),size = Airways_7_csec_a+size2,subset = !is.na(Airways_7_csec)), shape=22) +
#   geom_tippoint(aes(x = x+14, fill = log10(Airways_30_csec),size = Airways_30_csec_a+size2,subset = !is.na(Airways_30_csec)), shape=22) +
#   geom_tippoint(aes(x = x+15, fill = log10(Airways_90_csec),size = Airways_90_csec_a+size2,subset = !is.na(Airways_90_csec)), shape=22) +
# scale_fill_gradient2(low = 'red',high = 'darkgreen',midpoint = 0,mid = 'white',na.value = 'grey95',name = 'log10(OR)') +  

geom_tile(data = dfg3,aes(xx,y,fill = z))  + 
  scale_fill_gradient2(low = 'red',high = 'darkgreen',midpoint = 0,mid = 'white',na.value = 'grey95',name = 'weigted_OR') + 
  
  #geom_text(aes(label=label,angle=angle), hjust=-.5) +
  #geom_text(aes(label =node)) + 
  geom_rug(data = g3$data[g3$data$isTip,],sides = 'r', size = 3, aes(color = Family3)) + 
  scale_color_manual(values = cols) +
  geom_label(data = data.frame(x = max(g3$data$x) + c(4,12, 2,6,10,14),
                               y = max(g3$data$y)+c(14,14,rep(8,4)), 
                               lb = c('Fecal','Airways','Vag','Csec','Vag','Csec')), 
             aes(x,y,label = lb),size = 12) + 
  geom_text(data = data.frame(x = max(g3$data$x) + c(1,2,3,5,6,7,9,10,11,13,14,15),
                              y = max(g3$data$y)+3, 
                              lb = c(rep(c('1w','1m','1y'),2),rep(c('1w','1m','3m'),2))), 
            aes(x,y,label = lb),size = 8) + 
  theme(legend.position="right",legend.title=element_blank())


# g3
pdf('./phylotree_transfer_rect_family3.pdf',width = 20,height = 80)
g3 + xlim(c(0,56)) + guides(fill =  guide_legend(title = 'wOR', keywidth = 4, keyheight = 5),
                            color =  guide_legend(title = '', keywidth = 10, keyheight = 5, nrow = 5)) + 
  theme(legend.position = 'top',    legend.text=element_text(size=30))


dev.off()

```
